unstructured/ingest/interfaces.py

```diff
@@ -169,17 +178,19 @@ class ReadConfig(BaseConfig):
 
 @dataclass
 class EmbeddingConfig(BaseConfig):
-    api_key: str
+    provider: str
+    api_key: t.Optional[str] = None
     model_name: t.Optional[str] = None
 
     def get_embedder(self) -> BaseEmbeddingEncoder:
-        # TODO update to incorporate other embedder types once they exist
-        kwargs = {
-            "api_key": self.api_key,
-        }
+        kwargs = {}
+        if self.api_key:
+            kwargs["api_key"] = self.api_key
         if self.model_name:
             kwargs["model_name"] = self.model_name
-        return OpenAIEmbeddingEncoder(**kwargs)
+
+        cls = EMBEDDING_PROVIDER_TO_CLASS_MAP.get(self.provider, None)
```

Comment:
Rather than default this to `None` if not found, this should raise an error.

Reply:
Agreed, addressed with [raise error when wrong provider name is provided](https://github.com/Unstructured-IO/unstructured/pull/1918/commits/9cac06ab1f059a071e1c190393a5a444105596b3)